node-mixin/rules/rules.libsonnet

```diff
@@ -0,0 +1,121 @@
+{
+  prometheusRules+:: {
+    groups+: [
+      {
+        name: 'node.rules',
+        rules: [
+          {
+            // This rule gives the number of CPUs per node.
+            record: 'instance:node_num_cpu:sum',
+            expr: |||
+              count by (instance) (
+                sum by (instance, cpu) (
+                  node_cpu{%(nodeExporterSelector)s}
+                )
+              )
+            ||| % $._config,
+          },
+          {
+            // CPU utilisation is % CPU is not idle.
+            record: 'instance:node_cpu_utilisation:avg1m',
+            expr: |||
+              1 - avg by (instance) (
+                rate(node_cpu{%(nodeExporterSelector)s,mode="idle"}[1m])
+              )
+            ||| % $._config,
+          },
+          {
+            // CPU saturation is 1min avg run queue length / number of CPUs.
+            // Can go over 100%.  >100% is bad.
+            record: 'instance:node_cpu_saturation_load1:',
+            expr: |||
+              sum by (instance) (
+                node_load1{%(nodeExporterSelector)s}
+              )
+              /
+              instance:node_num_cpu:sum
+            ||| % $._config,
+          },
+          {
+            // Available memory per node
+            record: 'instance:node_memory_bytes_available:sum',
+            expr: |||
+              sum by (instance) (
+                (node_memory_MemFree{%(nodeExporterSelector)s} + node_memory_Cached{%(nodeExporterSelector)s} + node_memory_Buffers{%(nodeExporterSelector)s})
+              )
+            ||| % $._config,
+          },
+          {
+            // Total memory per node
+            record: 'instance:node_memory_bytes_total:sum',
```

Comment:
The total here is confusing, this is not a counter.

It's generally not a good idea to have a recording rule that's just renaming a metric name.

Reply:
Removed.